### ISSUE TITLE
Galaxy downloading with native Linux titles ignores split logic

### ISSUE THREAD

**Issue by shmerl** [Sude-/lgogdownloader#163]
I just tested updating x4 foundations with Galaxy simulation as follows:

```
lgogdownloader --galaxy-platform l --subdir-galaxy-install x4_foundations --galaxy-install 1588366064
```
And I noticed that it downloaded a few split files like

```
01.dat.split00
01.dat.split01
```
which normal installer concatenates into single `01.dat`. I suppose this split / cat logic is currently a missing feature?

**Comment by Sude-** [comment]
I didn't know that split files were even a thing as I hadn't come across them yet.
Have you seen them in other installers?

I've started writing some preliminary code for handling split files.
I'll try to get some code written in the next few days and create new branch for you to test if it works.

**Comment by shmerl** [comment]
Yes, I've sen them for example in TW2. Basically, when files are above certain size, Mojosetup can't handle them for some reason (may be something is stuck at 32-bit there), and GOG split files as a hack for such cases.

**Comment by shmerl** [comment]
In cases when split files occur, you'll find in `support` subdirectory something like this:
```
split_files 
```
In case of TW2 it contains:

./game/CookedPC/pack0.dzip

In case of X4: Foundations:
```
./game/01.dat
```
In case of The Bard's Tale IV:

```
./game/BardsTale4/Content/Paks/pakchunk2-LinuxNoEditor.pak
./game/BardsTale4/Content/Paks/pakchunk0-LinuxNoEditor.pak
```

And so on. May be you can use that `split_files` as the source of info, which files require concatenation.

See also GOG's scripts like

```
postinst.sh 
preuninst.sh
```

**Comment by Sude-** [comment]
I own TW2 so I have at least one game I can test my code on.

**Comment by Sude-** [comment]
I've created [splitfiles branch](/Sude-/lgogdownloader/tree/splitfiles) which has support for split files.
Let me know if there are any issues with it.

I decided not to read split files from ```data/noarch/support/split_files``` but detect them using regex.
Hopefully this doesn't cause any issues.

It deletes the split file parts after combining so user doesn't need to keep lots of unnecessary data around.
I also made it able to detect if previously combined base file is the same as split file. Meaning that it can do very basic version checking on split files.

**Comment by shmerl** [comment]
> I decided not to read split files from data/noarch/support/split_files but detect them using regex.
Hopefully this doesn't cause any issues.

It's probably better to read data/noarch/support/split_files, to avoid potential name conflicts or any future changes to the naming scheme. Implicit detection method will probably work most of the time, but it's just less reliable.

**Comment by Sude-** [comment]
d386bde improves split file support by reading the base file names from ```data/noarch/support/split_files```

**Comment by Sude-** [comment]
Split file support is now merged to master c9f9882
